pkg/trace/stats/concentrator_test.go

```diff
@@ -381,42 +424,118 @@ func TestConcentratorSublayersStatsCounts(t *testing.T) {
 		"query|duration|env:none,resource:resource1,service:A1":                                           2000,
 		"query|duration|env:none,resource:resource2,service:A2":                                           1000,
 		"query|duration|env:none,resource:resource3,service:A2":                                           1000,
 		"query|duration|env:none,resource:resource4,service:A3":                                           40,
 		"query|duration|env:none,resource:resource6,service:A3":                                           30,
-		"query|errors|env:none,resource:resource1,service:A1":                                             0,
-		"query|errors|env:none,resource:resource2,service:A2":                                             0,
-		"query|errors|env:none,resource:resource3,service:A2":                                             0,
-		"query|errors|env:none,resource:resource4,service:A3":                                             0,
-		"query|errors|env:none,resource:resource6,service:A3":                                             0,
+		"nested_op|duration|env:none,resource:resource1,service:A1":                                       200,
 		"query|hits|env:none,resource:resource1,service:A1":                                               1,
 		"query|hits|env:none,resource:resource2,service:A2":                                               1,
 		"query|hits|env:none,resource:resource3,service:A2":                                               1,
 		"query|hits|env:none,resource:resource4,service:A3":                                               1,
 		"query|hits|env:none,resource:resource6,service:A3":                                               1,
+		"nested_op|hits|env:none,resource:resource1,service:A1":                                           1,
+		"query|errors|env:none,resource:resource1,service:A1":                                             0,
+		"query|errors|env:none,resource:resource2,service:A2":                                             0,
+		"query|errors|env:none,resource:resource3,service:A2":                                             0,
+		"query|errors|env:none,resource:resource4,service:A3":                                             0,
+		"query|errors|env:none,resource:resource6,service:A3":                                             0,
+		"nested_op|errors|env:none,resource:resource1,service:A1":                                         0,
 	}
+	countsEq(assert, expectedCountValByKey, receivedCounts)
+}
 
-	// verify we got all counts
-	assert.Equal(len(expectedCountValByKey), len(receivedCounts), "GOT %v", receivedCounts)
-	// verify values
-	for key, val := range expectedCountValByKey {
-		count, ok := receivedCounts[key]
-		assert.True(ok, "%s was expected from concentrator", key)
-		assert.Equal(val, int64(count.Value), "Wrong value for count %s", key)
-	}
+func TestConcentrator_AddNow(t *testing.T) {
```

Comment:
Added simple test cases for the aggregation behavior, separate from any timestamp management.

Reply:
Can we please remove the underscore from the name?